pkg/operator/starter.go

```diff
@@ -745,3 +757,33 @@ func withVolumeGroupSnapshotWebhook(enabled bool) dc.DeploymentHookFunc {
 		return nil
 	}
 }
+
+func extractOperatorSpec(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
+	castObj := &operatorv1.CSISnapshotController{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to CSISnapshotController: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractCSISnapshotController(castObj, fieldManager)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract fields for %q: %w", fieldManager, err)
+	}
+	if ret.Spec == nil {
+		return nil, nil
+	}
+	return &ret.Spec.OperatorSpecApplyConfiguration, nil
+}
+func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
+	castObj := &operatorv1.CSISnapshotController{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to CSISnapshotController: %w", err)
+	}
+	ret, err := applyoperatorv1.ExtractCSISnapshotController(castObj, fieldManager)
```

Comment:
s/ExtractCSISnapshotController/ExtractCSISnapshotControllerStatus

Reply:
Thank you